pkgs/desktops/deepin/dde-daemon/default.nix

```diff
@@ -78,6 +94,9 @@ buildGoPackage rec {
   installPhase = ''
     make install PREFIX="$out" -C go/src/${goPackagePath}
     remove-references-to -t ${go} $out/lib/deepin-daemon/*
+    wrapProgram $out/lib/deepin-daemon/dde-session-daemon \
```

Comment:
Why can't you make use of `wrapGAppsHook`?

Reply:
I do not know how to use `wrapGAppsHook` with files outside of `$out/bin` and `$out/libexec` dirs.

Reply:
Right, I don't think we can actually do that currently.

Perhaps try
```
postFixup = ''
    makeWrapper $out/lib/deepin-daemon/dde-session-daemon "''${gappsWrapperArgs[@]}"
'';
```

That should produce a wrapper with all the arguments `wrapGAppsHook` would have gave `wrapProgram`.

Reply:
Using `makeWrapper` as you suggested is not working. It gives an error. After adding `wrapGAppsHook` to `nativeBuildInputs` and adding the following to the `installPhase`
```
    echo makeWrapper $out/lib/deepin-daemon/dde-session-daemon "''${gappsWrapperArgs[@]}"
    makeWrapper $out/lib/deepin-daemon/dde-session-daemon "''${gappsWrapperArgs[@]}"
```
I am getting the following message:
```
makeWrapper /nix/store/d77bb9n4bk46h10dk8xz7xbyj49s5f7l-dde-daemon-3.24.1/lib/deepin-daemon/dde-session-daemon --prefix GIO_EXTRA_MODULES : /nix/store/6png1cvn0jvrzkhjrxi4nvbjsff8y8af-dconf-0.30.1-lib/lib/gio/modules --prefix GIO_EXTRA_MODULES : /nix/store/6png1cvn0jvrzkhjrxi4nvbjsff8y8af-dconf-0.30.1-lib/lib/gio/modules
dirname: unrecognized option '--prefix'
Try 'dirname --help' for more information.
mkdir: cannot create directory '': No such file or directory
```

Reply:
This should work
```diff
diff --git a/pkgs/desktops/deepin/dde-daemon/default.nix b/pkgs/desktops/deepin/dde-daemon/default.nix
index 7271ce8c0b2..2690547f0a9 100644
--- a/pkgs/desktops/deepin/dde-daemon/default.nix
+++ b/pkgs/desktops/deepin/dde-daemon/default.nix
@@ -4,7 +4,7 @@
   deepin-wallpapers, deepin-desktop-base, alsaLib, glib, gtk3,
   libgudev, libinput, libnl, librsvg, linux-pam, networkmanager,
   pulseaudio, xorg, python3Packages, hicolor-icon-theme, glibc,
-  tzdata, go, deepin, makeWrapper }:
+  tzdata, go, deepin, makeWrapper, wrapGAppsHook }:
 
 buildGoPackage rec {
   name = "${pname}-${version}";
@@ -45,6 +45,7 @@ buildGoPackage rec {
     python3Packages.python
     makeWrapper
     deepin.setupHook
+    wrapGAppsHook
   ];
 
   buildInputs = [
@@ -94,11 +95,15 @@ buildGoPackage rec {
   installPhase = ''
     make install PREFIX="$out" -C go/src/${goPackagePath}
     remove-references-to -t ${go} $out/lib/deepin-daemon/*
-    wrapProgram $out/lib/deepin-daemon/dde-session-daemon \
-      --prefix XDG_DATA_DIRS : "$GSETTINGS_SCHEMAS_PATH"
     searchHardCodedPaths $out
   '';
 
+  postFixup = ''
+    for binary in $out/lib/deepin-daemon/*; do
+      wrapProgram $binary "''${gappsWrapperArgs[@]}"
+    done
+  '';
+
   passthru.updateScript = deepin.updateScript { inherit name; };
 
   meta = with stdenv.lib; {
```

Note that it has to be done in `postFixup`.